src/resmom/linux/alps.c

```diff
@@ -4165,7 +4165,7 @@ inventory_to_vnodes(basil_response_t *brp)
 {
 	extern int internal_state_update;
 	extern int num_acpus;
-	extern ulong totalmem;
+	extern unsigned long totalmem;
```

Comment:
can we use typedef instead?
typedef unsigned long ulong;

Reply:
We can, but I surveyed the code base and `unsigned long` is used almost everywhere, so I figured it makes more sense to just remove the last few instances of `ulong`.

Reply:
agreed! If you include it in the correct header file, ulong can still be added in the future without breaking mac. As of now, I don't see a CI pipeline for mac.

Reply:
I'm not quite sure if there's a way to guard against it generally, but other things could also be added that break the macOS build. I'm not too worried about this right now as, without CI, there's no way to guarantee the build always work but at least we can minimize any patches required to package OpenPBS for Mac (as I intend to do).